app/assets/javascripts/project_find_file.js.coffee

```diff
@@ -0,0 +1,121 @@
+class @ProjectFindFile
+  constructor: (@element, @options)->
+    @filePaths = {}
+    @inputElement = @element.find(".file-finder-input")
+
+    # init event
+    @initEvent()
+
+    # focus text input box
+    @inputElement.focus()
+
+    # load file list
+    @load(@options.url)
+
+  # init key event
+  initEvent: ->
+    @inputElement.off "keyup"
+    @inputElement.on "keyup", (event) =>
+      target = $(event.target)
+      value = target.val()
+
+      if value != target.data("oldValue")
+        target.data("oldValue", value)
+        @findFile()
+        @element.find('tr.tree-item').eq(0).addClass("selected").focus()
+
+    @element.find(".tree-content-holder .tree-table").on 'click', (e) ->
+      if (e.target.nodeName != "A")
+        path = $('.tree-item-file-name a', this).attr('href')
+        if path then location.href = path
+
+  # find file
+  findFile: ->
+    searchTxt = @inputElement.val()
+    result = if searchTxt then fuzzaldrinPlus.filter(@filePaths, searchTxt) else @filePaths
+    @renderList result, searchTxt
+
+
+  # files pathes load
+  load: (url) ->
+    $.ajax
+      url: url
+      method: "get"
+      dataType: "json"
+      success: ((data) ->
+        @element.find(".loading").hide()
+        @filePaths = data
+        @findFile()
+        @element.find('#files-slider tr.tree-item').eq(0).addClass("selected").focus()
+      ).bind(this)
+
+  # render result
+  renderList: (filePaths, searchTxt) ->
+    $(".tree-table > tbody").empty()
+
+    for filePath, i in filePaths
+      break if i > 19
+      highlightFilePath = filePath
+
+      if searchTxt
+        matches = fuzzaldrinPlus.match(filePath, searchTxt)
+        highlightFilePath = highlighter filePath, matches
+
+      blobItemUrl = "#{@options.blobUrlTemplate}/#{filePath}"
+      html = @makeHtml highlightFilePath, blobItemUrl
+      $(".tree-table > tbody").append(html)
+
+  # highlight text(awefwbwgtc -> <b>a</b>wefw<b>b</b>wgt<b>c</b> )
+  highlighter = (text, matches) ->
+    lastIndex = 0
+    highlightText = ""
+    matchedChars = []
+
+    for matchIndex in matches
+      unmatched = text.substring(lastIndex, matchIndex)
+
+      if unmatched
+        highlightText += matchedChars.join('').bold() if matchedChars.length
+        matchedChars = []
+        highlightText += unmatched
+
+      matchedChars.push(text[matchIndex])
+      lastIndex = matchIndex + 1
+
+    highlightText += matchedChars.join('').bold() if matchedChars.length
+    highlightText += text.substring(lastIndex)
+
+  # make tbody row html
+  makeHtml: (filePath, blobItemUrl) ->
+    return $("<tr class='tree-item'><td class='tree-item-file-name'><i class='fa fa-file-text-o fa-fw'></i><span class='str-truncated'><a href=#{blobItemUrl}>#{filePath}</a></span></td></tr>")

```

Comment:
To prevent XSS when `blobItemUrl` or `filePath` contain something that looks like HTML/JS, Build the `span` element like you do now, and then create the link like `$('<a>').attr('href', blobItemUrl).text(filePath)` and add it to the span using jQuery.